RetailCoder.VBE/Inspections/QuickFixes/AssignedByValParameterQuickFix.cs

```diff
@@ -0,0 +1,186 @@
+                                    || line.Contains(NameAsSubOrFunctionParamLast());
+        }
+
+        private string NameAsLeftHandSide() { return _target.IdentifierName + " "; }
+        private string NameAsRightHandSide() { return " " + _target.IdentifierName; }
+        private string NameAsObjectMethodOrAccessorCall() { return " " + _target.IdentifierName + "."; }
+        private string NameAsSubOrFunctionParam() { return _target.IdentifierName + ","; }
+        private string NameAsSubOrFunctionParamFirst() { return "(" + _target.IdentifierName; }
+        private string NameAsSubOrFunctionParamLast() { return _target.IdentifierName + ")"; }
+
+
+        private string buildLocalCopyDeclaration()
+        {
+            return Tokens.Dim + " " + _localCopyVariableName + " " + Tokens.As 
+                + " " + _target.AsTypeName;
+        }
+        private string buildLocalCopyAssignment()
+        {
+            return (ValueTypes.Contains(_target.AsTypeName) ? string.Empty : Tokens.Set + " ") 
+                + _localCopyVariableName + " = " + _target.IdentifierName; ;
+        }
+        private string[] GetModuleLines()
+        {
+            var module = Selection.QualifiedName.Component.CodeModule;
+            var lines = module.GetLines(1, module.CountOfLines);
+            string[] newLine = new string[] { "\r\n" };
+            return lines.Split(newLine, StringSplitOptions.None);
+        }
+        private string[] GetMethodLines()
+        {
+            int zbIndex; //zero-based index
+            zbIndex = Selection.Selection.StartLine - 1;
+            string[] allLines = GetModuleLines();
+
+            string endStatement = GetEndOfScopeStatementForDeclaration(allLines[zbIndex]);
+
+            bool IsInScope = true;
+            var codeBlockLines = new List<string>();
+            for ( ; IsInScope && zbIndex < allLines.Count(); zbIndex++)
+            {
+                codeBlockLines.Add(allLines[zbIndex]);
+                IsInScope = !allLines[zbIndex].Contains(endStatement);
+            }
+            return codeBlockLines.ToArray();
+        }
+        private string GetEndOfScopeStatementForDeclaration(string declaration)
+        {
+            return declaration.Contains("Sub ") ? "End Sub" : "End Function";
+        }
+        private static readonly IReadOnlyList<string> ValueTypes = new[]
+        {
+            Tokens.Boolean,
+            Tokens.Byte,
+            Tokens.Currency,
+            Tokens.Date,
+            Tokens.Decimal,
+            Tokens.Double,
+            Tokens.Integer,
+            Tokens.Long,
+            Tokens.LongLong,
+            Tokens.Single,
+            Tokens.String,
+            Tokens.Variant
+        };
+        private string AutoSuggestedName()
+        {
+            return "local" + CapitalizeFirstLetter(_target.IdentifierName);
+        }
+        private string CapitalizeFirstLetter(string name)
```

Comment:
Thanks!  I'll make this update.  On longer variable names, .Capitalize() appears to change all subsequent capital letters in the remainder of the name to lower case (foo**B**ar becomes localFoo**b**ar).  Since this is simply an auto-suggested name, the locale awareness is probably more important than retaining the internal capital letters....the user is welcome to put them back in.